apis_v1/views/views_donation.py

```diff
@@ -229,3 +229,65 @@ def validate_coupon_for_api_view(request):
         }
 
     return HttpResponse(json.dumps(json_data), content_type='application/json')
+
+def create_new_plan_for_api_view(request):
+    coupon_code = request.GET.get('couponCode')
+    plan_type_enum = request.GET.get('planTypeEnum')
+    hidden_plan_comment = request.GET.get('hiddenPlanComment')
+    coupon_applied_message = request.GET.get('couponAppliedMessage')
+    list_price_monthly_credit = request.GET.get('listPriceMonthlyCredit')
+    discounted_price_monthly_credit = request.GET.get('discountedPriceMonthlyCredit')
+    features_provided_bitmap = request.GET.get('featuresProvidedBitmap')
+    coupon_expires_date = request.GET.get('couponExpiresDate', None)
+    if len(coupon_expires_date) is 0:
+        coupon_expires_date = None
+    print("create_new_plan_for_api_view, plan_type_enum: " + plan_type_enum + ", coupon_code: " + coupon_code)
+    plan_on_stage = 0
+
+    voter_device_id = get_voter_device_id(request)  # We standardize how we take in the voter_device_id
```

Comment:
Please also restrict this to 'admin' only, using this code (I will add a 'account_manager' classification later):
```
    # admin, partner_organization, political_data_manager, political_data_viewer, verified_volunteer
    authority_required = {'admin'}
    if not voter_has_authority(request, authority_required):
```

Reply:
Done!